BOM is not usable with spring-boot-starter-parent

### Version

6.1.0

### What happened?

When adding `apache-jena-libs` or 'jena-bom' to a Maven paren tom project , the Spring version is overriden ,y apache shiro import.
This makes  the current BOM/parent setup effectively unusable in any project that needs to control its own Spring version.

Steps to Reproduce:

Create a maven project and import

```
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">

   <modelVersion>4.0.0</modelVersion>

   <parent>
      <groupId>org.springframework.boot</groupId>
      <artifactId>spring-boot-starter-parent</artifactId>
      <version>4.1.0</version>
   </parent>
   <dependencyManagement>
      <dependencies>
       <dependency>
               <groupId>org.apache.jena</groupId>
               <artifactId>jena-bom</artifactId>
               <version>${jena.version}</version>
               <type>pom</type>
            <scope>import</scope>
         </dependency>
```

Run mvn dependency:tree or mvn help:effective-pom.

and see that spring web ist 5.x.x instead of 7.x.x

### Relevant output and stacktrace

```shell

```

### Are you interested in making a pull request?

_No response_

Jip without import it's not possible to use it as dependency management section.

See new commit

just run clean install

```
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[ERROR] 'dependencies.dependency.version' for org.apache.jena:jena-rdfpatch:jar is missing. @ line 25, column 19
 @ 
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]   
[ERROR]   The project jena-test:jena-test-module:DEV-SNAPSHOT (/jena-test/jena-test-module/pom.xml) has 1 error
[ERROR]     'dependencies.dependency.version' for org.apache.jena:jena-rdfpatch:jar is missing. @ line 25, column 19
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
```

--- a/jena-bom/pom.xml
+++ b/jena-bom/pom.xml
@@ -34,6 +34,37 @@
     <version>6.3.0-SNAPSHOT</version>
   </parent>
 
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>flatten-maven-plugin</artifactId>
+        <configuration>
+          <flattenMode>bom</flattenMode>
+          <pomElements>
+            <properties>remove</properties>
+          </pomElements>
+        </configuration>
+        <executions>
+          <execution>
+            <id>flatten</id>
+            <phase>process-resources</phase>
+            <goals>
+              <goal>flatten</goal>
+            </goals>
+          </execution>
+          <execution>
+            <id>flatten.clean</id>
+            <phase>clean</phase>
+            <goals>
+              <goal>clean</goal>
+            </goals>
+          </execution>
+        </executions>
+      </plugin>
+    </plugins>
+  </build>
+
   <dependencyManagement>
     <dependencies>
 

--- a/pom.xml
+++ b/pom.xml
@@ -138,6 +138,7 @@
     <ver.plugin.enforcer>3.6.3</ver.plugin.enforcer>
     <ver.plugin.cyclonedx>2.9.2</ver.plugin.cyclonedx>
     <ver.plugin.spdx>0.7.0</ver.plugin.spdx>
+    <ver.plugin.flatten>1.8.0</ver.plugin.flatten>
 
   </properties>
 
@@ -1081,6 +1082,11 @@
           <version>${ver.plugin.shade}</version>
         </plugin>
 
+        <plugin>
+          <groupId>org.codehaus.mojo</groupId>
+          <artifactId>flatten-maven-plugin</artifactId>
+          <version>${ver.plugin.flatten}</version>
+        </plugin>
       </plugins>
     </pluginManagement>
   </build>

